main/input_default.cpp

```diff
@@ -1338,7 +1338,10 @@ String InputDefault::get_joy_button_string(int p_button) {
 
 int InputDefault::get_joy_button_index_from_string(String p_button) {
 	for (int i = 0; i < JOY_BUTTON_MAX; i++) {
-		if (p_button == _buttons[i]) {
+		if (_buttons[i] == nullptr) {
+			break;
```

Comment:
Might be worth adding an explicit error message in the `ERR_FAIL_V` below:
```
ERR_FAIL_V_MSG(-1, vformat(R("Could not find a button index matching the string "%s"."), p_button));
``` 
And same for axis.

Then please squash the commits together and it should be good to merge.

Reply:
Oh I totally agree with you. I added it and squash the commits 😊